src/Concerns/ManagesTransactions.php

```diff
@@ -88,6 +92,8 @@ public function transaction(Closure $callback, $attempts = 1, array $options = [
         $callbackResult = null;
         $throwable      = null;
 
+        $this->transactionsManager?->begin($this->getName(), 0);
```

Comment:
Unfortunately have to call this stuff here manually again, unlike what the Laravel trait does (which is to call its own `beginTransaction` method here), as we are using the `with_transaction` helper function and passing the callback. The helper function, in turn, calls `WithTransaction::execute` which handles starting and committing the actual transaction in the session.

Calling `beginTransaction` here instead of calling `DatabaseTransactionsManager::begin` directly would therefore result in attempting to start the transaction twice.